dspace-api/src/main/java/org/dspace/importer/external/datacite/DataCiteImportMetadataSourceServiceImpl.java

```diff
@@ -9,43 +9,50 @@
 
 import java.util.ArrayList;
 import java.util.Collection;
-import java.util.LinkedHashMap;
+import java.util.HashMap;
+import java.util.Iterator;
 import java.util.List;
-import java.util.concurrent.Callable;
+import java.util.Map;
 import javax.el.MethodNotFoundException;
-import javax.ws.rs.client.Client;
-import javax.ws.rs.client.ClientBuilder;
-import javax.ws.rs.client.Invocation;
 import javax.ws.rs.client.WebTarget;
-import javax.ws.rs.core.Response;
 
-import com.google.gson.Gson;
-import com.jayway.jsonpath.JsonPath;
-import com.jayway.jsonpath.ReadContext;
-import net.minidev.json.JSONArray;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.apache.commons.lang3.StringUtils;
-import org.apache.http.client.methods.HttpGet;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.dspace.content.Item;
 import org.dspace.importer.external.datamodel.ImportRecord;
 import org.dspace.importer.external.datamodel.Query;
 import org.dspace.importer.external.exception.MetadataSourceException;
+import org.dspace.importer.external.liveimportclient.service.LiveImportClient;
 import org.dspace.importer.external.service.AbstractImportMetadataSourceService;
 import org.dspace.importer.external.service.DoiCheck;
 import org.dspace.importer.external.service.components.QuerySource;
+import org.springframework.beans.factory.annotation.Autowired;
 
 /**
  * Implements a data source for querying Datacite
  * Mainly copied from CrossRefImportMetadataSourceServiceImpl.
  *
  * optional Affiliation informations are not part of the API request.
  * https://support.datacite.org/docs/can-i-see-more-detailed-affiliation-information-in-the-rest-api
- * 
+ *
  * @author Pasquale Cavallo (pasquale.cavallo at 4science dot it)
  * @author Florian Gantner (florian.gantner@uni-bamberg.de)
  *
  */
 public class DataCiteImportMetadataSourceServiceImpl
-    extends AbstractImportMetadataSourceService<String> implements QuerySource {
+        extends AbstractImportMetadataSourceService<String> implements QuerySource {
+    private final static Logger log = LogManager.getLogger();
+
+    @Autowired
+    private LiveImportClient liveImportClient;
+
+    private final int timeoutMs = 180000;
+
+    private final String url = "https://api.datacite.org/dois/";
 
     private WebTarget webTarget;
```

Comment:
still some legacy and unused import

Reply:
Thanks @floriangantner, will be fixed asap!

Reply:
Has been fixed! :grin: